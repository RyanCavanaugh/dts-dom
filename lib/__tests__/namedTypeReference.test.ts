import * as dom from "../index"

describe("an emitted named reference", () => {
    describe("without type arguments", () => {
        it("matches the snapshot", () => {
            let typeReference = dom.create.namedTypeReference(
                "MyInterface"
            );
            let alias = dom.create.alias('Alias', typeReference);

            expect(dom.emit(alias)).toMatchSnapshot();
        });
    });

    describe("with single type argument", () => {
        it("matches the snapshot", () => {
            let typeReference = dom.create.namedTypeReference(
                "Promise"
            );
            typeReference.typeArguments.push(dom.type.string);
            let alias = dom.create.alias('Alias', typeReference);

            expect(dom.emit(alias)).toMatchSnapshot();
        });
    });
    
    describe("with multiple type arguments", () => {
        it("matches the snapshot", () => {
            let typeReference = dom.create.namedTypeReference(
                "Generator"
            );
            typeReference.typeArguments.push(dom.type.string);
            typeReference.typeArguments.push(dom.type.number);
            let alias = dom.create.alias('Alias', typeReference);

            expect(dom.emit(alias)).toMatchSnapshot();
        });
    });
});
