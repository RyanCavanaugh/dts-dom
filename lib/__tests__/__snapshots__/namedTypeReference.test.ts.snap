// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`an emitted named reference with multiple type arguments matches the snapshot 1`] = `
declare type Alias = Generator<string, number>;


`;

exports[`an emitted named reference with single type argument matches the snapshot 1`] = `
declare type Alias = Promise<string>;


`;

exports[`an emitted named reference without type arguments matches the snapshot 1`] = `
declare type Alias = MyInterface;


`;
